graphql_compiler/tests/test_post_processing.py

```diff
@@ -0,0 +1,221 @@
+        ]
+
+        post_process_mssql_folds(self.mssql_schema_info, graphql_query, query_output)
+        self.assertEqual(query_output, expected_result)
+
+    def test_convert_ampersand_encodings(self):
+        """Test ampersand, less than, and greater than are correctly decoded."""
+        query_output = [
+            {
+                "child_names": "name with a &amp; (ampersand)|"
+                "name with a &gt; (greater than)|"
+                "name with a &lt; (less than)|"
+                "&amp;many&amp;&gt;ampersand&lt;escaped&lt;&lt;&gt;name&amp;",
+            }
+        ]
+        graphql_query = """{
+                            Animal {
+                                in_Animal_ParentOf @fold{
+                                    name @output(out_name: "child_names")
+                                }
+                            }
+                        }"""
+
+        expected_result = [
+            {
+                "child_names": [
+                    "name with a & (ampersand)",
+                    "name with a > (greater than)",
+                    "name with a < (less than)",
+                    "&many&>ampersand<escaped<<>name&",
+                ]
+            }
+        ]
+
+        post_process_mssql_folds(self.mssql_schema_info, graphql_query, query_output)
+        self.assertEqual(query_output, expected_result)
+
+    def test_convert_hex_encodings(self):
+        """Test HTML hex encodings are properly decoded."""
+        query_output = [
+            {
+                "child_names": "name with a &#x06; (acknowledge)|"
+                "&#x0B;many&#x06;hex&#x0F;&#x07;name&#x08;",
+            }
+        ]
+        graphql_query = """{
+                                    Animal {
+                                        in_Animal_ParentOf @fold{
+                                            name @output(out_name: "child_names")
+                                        }
+                                    }
+                                }"""
+
+        expected_result = [
+            {"child_names": ["name with a \x06 (acknowledge)", "\x0Bmany\x06hex\x0F\x07name\x08",]}
+        ]
+
+        post_process_mssql_folds(self.mssql_schema_info, graphql_query, query_output)
+        self.assertEqual(query_output, expected_result)
+
+    def test_convert_basic_decimal(self):
+        """Test basic XML path encoding for decimals is correctly decoded."""
+        query_output = [{"child_net_worths": "500|1000|400|~",}]
+        graphql_query = """{
+            Animal {
+                in_Animal_ParentOf @fold{
+                    net_worth @output(out_name: "child_net_worths")
+                }
+            }
+        }"""
+
+        expected_result = [{"child_net_worths": [500, 1000, 400, None,],}]
+
+        post_process_mssql_folds(self.mssql_schema_info, graphql_query, query_output)
+        self.assertEqual(query_output, expected_result)
+
+    def test_convert_basic_datetime(self):
```

Comment:
I'd actually want to see a test with datetimes. Do we enforce a particular datetime format when encoding the datetimes? Do we include timezone information?